skyrim-platform/src/platform_se/skyrim_platform/GameEventSinks.cpp

```diff
@@ -665,3 +732,1297 @@ RE::BSEventNotifyControl GameEventSinks::ProcessEvent(
+    return RE::BSEventNotifyControl::kContinue;
+  }
+
+  auto ref = event->reference;
+  auto refId = ref ? ref->formID : 0;
+
+  SkyrimPlatform::GetSingleton().AddUpdateTask([ref, refId] {
+    auto obj = JsValue::Object();
+
+    auto refLocal = RE::TESForm::LookupByID(refId);
+
+    if (refId == 0) {
+      obj.SetProperty("reference", JsValue::Null());
+    } else {
+      if (refLocal == nullptr || refLocal != ref) {
+        return;
+      }
+      obj.SetProperty("reference", CreateObject("ObjectReference", refLocal));
+    }
+
+    EventsApi::SendEvent("niNodeUpdate", { JsValue::Undefined(), obj });
+  });
+
+  return RE::BSEventNotifyControl::kContinue;
+}
+
+RE::BSEventNotifyControl GameEventSinks::ProcessEvent(
+  const SKSE::ModCallbackEvent* event,
+  RE::BSTEventSource<SKSE::ModCallbackEvent>* eventSource)
+{
+  if (event == nullptr) {
+    return RE::BSEventNotifyControl::kContinue;
+  }
+
+  auto eventName = event->eventName;
+  auto numArg = event->numArg;
+  auto strArg = event->strArg;
+  auto sender = event->sender;
+  auto senderId = sender ? sender->formID : 0;
+
+  SkyrimPlatform::GetSingleton().AddUpdateTask(
+    [eventName, numArg, strArg, sender, senderId] {
+      auto obj = JsValue::Object();
+
+      auto senderLocal = RE::TESForm::LookupByID(senderId);
+
+      if (senderLocal == nullptr || senderLocal != sender) {
+        return;
+      }
+
+      obj.SetProperty("sender", CreateObject("Form", senderLocal));
+      obj.SetProperty("eventName",
+                      JsValue::String(static_cast<std::string>(eventName)));
+      obj.SetProperty("strArg",
+                      JsValue::String(static_cast<std::string>(strArg)));
+      obj.SetProperty("numArg", JsValue::Double(numArg));
+
+      EventsApi::SendEvent("modEvent", { JsValue::Undefined(), obj });
+    });
+
+  return RE::BSEventNotifyControl::kContinue;
+}
+
+RE::BSEventNotifyControl GameEventSinks::ProcessEvent(
+  const RE::ItemCrafted::Event* event,
+  RE::BSTEventSource<RE::ItemCrafted::Event>* eventSource)
+{
+  if (event == nullptr) {
+    return RE::BSEventNotifyControl::kContinue;
+  }
+
+  auto form = event->item;
+  auto formId = form ? form->formID : 0;
+  auto weaponAlchemyFlag = event->unk08;
+  auto enchantingFlag = event->unk09;
+
+  SkyrimPlatform::GetSingleton().AddUpdateTask(
+    [form, formId, weaponAlchemyFlag, enchantingFlag] {
+      auto obj = JsValue::Object();
+
+      auto formLocal = RE::TESForm::LookupByID(formId);
+
+      if (formLocal == nullptr || formLocal != form) {
+        return;
+      }
+
+      obj.SetProperty("item", CreateObject("Form", formLocal));
+
+      if (weaponAlchemyFlag) {
+        EventsApi::SendEvent("itemCrafted", { JsValue::Undefined(), obj });
+      }
+      if (enchantingFlag) {
+        EventsApi::SendEvent("itemEnchanted", { JsValue::Undefined(), obj });
+      }
+    });
+
+  return RE::BSEventNotifyControl::kContinue;
+}
+
+RE::BSEventNotifyControl GameEventSinks::ProcessEvent(
```

Comment:
Same. I don't like the current interface. We better discuss this item in a separate PR.